monitoring/uss_qualifier/scenarios/astm/netrid/common/aggregate_checks.py

```
@@ -103,8 +103,35 @@ def run(self):
         self.end_test_step()
         self.end_test_case()
 
+        self.begin_test_case("Verify https is in use")
+        self.begin_test_step("Verify https is in use")
+
+        self._verify_https_everywhere()
+
+        self.end_test_step()
+        self.end_test_case()
+
         self.end_test_scenario()
 
+    def _verify_https_everywhere(self):
+
+        # TODO check if this is the correct way to do things:
+        #  Ie, if we want to define this as a capability and not a failing check if there is no https,
+        #  is it enough to declare a check only if everything went well?
+        #  alternatively, a failure with a severity of LOW might work too?
+
+        found_cleartext_query = False
+        for query in self._queries:
+            if query.request.url.startswith("http://"):
+                found_cleartext_query = True
+                self.record_note("https", f"query is not https: {query.request.url}")
+
+        if not found_cleartext_query:
+            self.check(
+                "All interactions happen over https",
+                self._queries_by_participant.keys(),
```

When possible, we should attribute failed checks to the specific participant failing them -- multiple participants for a single check means any one (or multiple) of the specified participants may have failed the check.  So, instead, we should iterate over participants and, for each participant, check whether all their queries use https.  If yes, pass; if no, fail; if no queries available, don't perform check.  (yes, in that case there will be multiple instances of the same check: one instance for each relevant participant)

---

Makes sense. This brings us back to the question of attributing queries to participants, and whether to do so by looking at the host-name or URL, or by tagging the requests where they are made:

We can probably set the `server_id` field for all queries. Though, what should we do if we run into a query that is not associated with an SP? I guess failing hard so that we uncover these when scenarios are developed?

---

Yes, I think server_id should be the means we can use to identify the participants.  I think we would want to use all data we have available to us to make the check, but I think we can actually simply ignore queries where we don't have a server ID.  For instance, if we added a test portion that interacted with a third entity other than a Service Provider or Display Provider and we happened not to populated server_id in our queries to them, I don't think that would justify failing hard.

That said, we may not be populating server_id in all the places that we should, so failing hard temporarily to make sure we've identified all those places seems fine.

---

I guess we could add a `dev` flag somewhere that will have more stringent internal requirements?

Or is leaving the code as-is fine and we remove the check at a later stage?

---

As discussed with @BenjaminPelletier offline: we'll leave this "debug" flag hardcoded for now. Once things are stable we can simply remove it.